// mime-validator/dist/classes/mimeType.js
/**
 * @class mimeTypes
 * Utility class that provides constants for various MIME types used in web development.
 * Categorizes MIME types into different groups such as text, image, audio, video, application, archive, font, XML, and binary.
 */
export class MimeType {
    /**
     * Returns an array of MIME types that are used for text-based content.
     *
     * @returns {string[]} An array of MIME types for text-based content, including plain text, HTML, CSS, and JavaScript.
     */
    static text() {
        return [
            this.PLAIN,
            this.HTML,
            this.CSS,
            this.JAVASCRIPT,
        ];
    }
    static image() {
        /**
         * Returns an array of MIME types that are used for image-based content.
         *
         * @returns {string[]} An array of MIME types for image-based content.
         */
        return [
            this.JPEG,
            this.PNG,
            this.GIF,
            this.SVG_XML,
            this.BMP,
            this.ICO,
            this.JLS,
            this.JP2,
            this.JPF,
            this.JPH,
            this.TIFF,
            this.WEBP
        ];
    }
    static audio() {
        /**
         * Returns an array of MIME types that are used for audio-based content.
         *
         * @returns {string[]} An array of audio MIME types.
         */
        return [
            this.AUDIO_MPEG,
            this.AUDIO_OGG,
            this.AUDIO_MP3,
            this.AUDIO_WAV,
            this.AUDIO_FLAC,
            this.AUDIO_AAC,
            this.AUDIO_FLV,
            this.AUDIO_WMA
        ];
    }
    static video() {
        /**
         * Returns an array of MIME types that are used for video-based content.
         *
         * @returns {string[]} An array of video MIME types, including `video/mp4`, `video/webm`, `video/ogg`, `video/x-msvideo`, `video/quicktime`, and `video/x-flv`.
         */
        return [
            this.VIDEO_MP4,
            this.VIDEO_WEBM,
            this.VIDEO_OGG,
            this.VIDEO_AVI,
            this.VIDEO_QUICKTIME,
            this.VIDEO_FLV
        ];
    }
    static application() {
        /**
         * Returns an array of MIME types that are commonly used for application-based content.
         *
         * @returns {string[]} An array of MIME types for application-based content.
         */
        return [
            this.PDF,
            this.MSWORD,
            this.POWERPOINT,
            this.EXCEL,
            this.JSON,
        ];
    }
    static archive() {
        /**
         * Returns an array of MIME types that are commonly used for archive-based content.
         *
         * @returns {string[]} An array of MIME types for archive-based content.
         */
        return [
            this.ZIP,
            this.GZIP,
            this.TAR,
            this.BZIP2,
            this.RAR,
            this.SEVENZIP,
            this.LHA
        ];
    }
    static font() {
        /**
         * Returns an array of MIME types that are used for font-based content.
         *
         * @returns {string[]} An array of MIME types for font-based content.
         */
        return [
            this.WOFF,
            this.WOFF2,
            this.SFNT,
            this.TTF,
            this.OTF,
            this.EOT,
            this.SVG
        ];
    }
    static xml() {
        /**
         * Returns an array of MIME types that are used for XML-based content.
         *
         * @returns {string[]} An array of MIME types for XML-based content.
         */
        return [
            this.XML,
            this.RSS_XML,
            this.ATOM_XML,
            this.TEXT_XML,
            this.XHTML_XML,
            this.SOAP_XML,
            this.XOP_XML,
        ];
    }
    static binary() {
        /**
         * Returns an array of MIME types that are commonly used for binary-based content.
         *
         * @returns {string[]} An array of MIME types for binary-based content.
         */
        return [
            this.BINARY,
            this.BSON,
            this.UBJSON,
            this.CBOR,
            this.MESSAGE_PACK,
            this.WASM
        ];
    }
}
MimeType.PLAIN = "text/plain";
MimeType.HTML = "text/html";
MimeType.CSS = "text/css";
MimeType.JAVASCRIPT = "text/javascript";
MimeType.JPEG = "image/jpeg";
MimeType.PNG = "image/png";
MimeType.GIF = "image/gif";
MimeType.SVG_XML = "image/svg+xml";
MimeType.BMP = "image/bmp";
MimeType.ICO = "image/x-icon";
MimeType.JLS = "image/jls";
MimeType.JP2 = "image/jp2";
MimeType.JPF = "image/jpx";
MimeType.JPH = "image/jph";
MimeType.TIFF = "image/tiff";
MimeType.WEBP = "image/webp";
MimeType.AUDIO_MPEG = "audio/mpeg";
MimeType.AUDIO_OGG = "audio/ogg";
MimeType.AUDIO_MP3 = "audio/mp3";
MimeType.AUDIO_WAV = "audio/wav";
MimeType.AUDIO_FLAC = "audio/flac";
MimeType.AUDIO_AAC = "audio/aac";
MimeType.AUDIO_FLV = "audio/x-flv";
MimeType.AUDIO_WMA = "audio/x-ms-wma";
MimeType.VIDEO_MP4 = "video/mp4";
MimeType.VIDEO_WEBM = "video/webm";
MimeType.VIDEO_OGG = "video/ogg";
MimeType.VIDEO_AVI = "video/x-msvideo";
MimeType.VIDEO_QUICKTIME = "video/quicktime";
MimeType.VIDEO_FLV = "video/x-flv";
MimeType.POWERPOINT = "application/vnd.ms-powerpoint";
MimeType.PDF = "application/pdf";
MimeType.MSWORD = "application/msword";
MimeType.EXCEL = "application/vnd.ms-excel";
MimeType.JSON = "application/json";
MimeType.ZIP = "application/zip";
MimeType.GZIP = "application/x-gzip";
MimeType.TAR = "application/x-tar";
MimeType.BZIP2 = "application/x-bzip2";
MimeType.RAR = "application/x-rar-compressed";
MimeType.SEVENZIP = "application/x-7z-compressed";
MimeType.LHA = "application/x-lzh-compressed";
MimeType.WOFF = "font/woff";
MimeType.WOFF2 = "font/woff2";
MimeType.TTF = "font/ttf";
MimeType.OTF = "font/otf";
MimeType.EOT = "application/vnd.ms-fontobject";
MimeType.SFNT = "application/font-sfnt";
MimeType.SVG = "image/svg+xml";
MimeType.XML = "application/xml";
MimeType.RSS_XML = "application/rss+xml";
MimeType.ATOM_XML = "application/atom+xml";
MimeType.TEXT_XML = "text/xml";
MimeType.XHTML_XML = "application/xhtml+xml";
MimeType.SOAP_XML = "application/soap+xml";
MimeType.XOP_XML = "application/xop+xml";
MimeType.BINARY = "application/octet-stream";
MimeType.BSON = "application/bson";
MimeType.UBJSON = "application/ubjson";
MimeType.CBOR = "application/cbor";
MimeType.MESSAGE_PACK = "application/msgpack";
MimeType.WASM = "application/wasm";
